benchmark/framework/parser/file_access.go

```diff
@@ -0,0 +1,177 @@
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package bparser
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"time"
+)
+
+var (
+	outputDir           = "./output/"
+	FileAccessDir       = outputDir + "file_access_logs/"
+	sociLogs            = outputDir + "soci-snapshotter-stderr"
+	containerdLogs      = outputDir + "containerd-stderr"
+	sociLogsFuseMessage = "FUSE operation"
+)
+
+type FileAccessPatterns struct {
+	ImageName           string         `json:"ImageName"`
+	ContainerStartTime  time.Time      `json:"containerStartTime"`
+	TotalOperationCount map[string]int `json:"TotalOperationCounts"`
+	Operations          []Operation    `json:"operations"`
+}
+
+type SociLog struct {
+	Msg string `json:"msg"`
+}
+type BaseOperation struct {
+	Level     string    `json:"level"`
+	Msg       string    `json:"msg"`
+	Operation string    `json:"operation"`
+	Path      string    `json:"path"`
+	Time      time.Time `json:"time"`
+}
+
+type Operation struct {
+	Operation                 string `json:"operation"`
+	Path                      string `json:"path"`
+	FirstAccessTimeAfterStart string `json:"firstAccessTimeAfterStart"`
+	Count                     int    `json:"count"`
+}
+
+func ParseFileAccesses(imageName string) error {
+	startTime, err := getTaskStartTime()
+	if err != nil {
+		return err
+	}
+	totalCounts := make(map[string]int)
+	fa := FileAccessPatterns{
+		ImageName:           imageName,
+		ContainerStartTime:  *startTime,
+		TotalOperationCount: totalCounts,
+	}
+	file, err := os.Open(sociLogs)
+	if err != nil {
+		return err
+	}
+
+	defer file.Close()
+
+	b, err := io.ReadAll(file)
+	if err != nil {
+		return err
+	}
+
+	logs := strings.Split(string(b), "\n")
+
+	m := make(map[string]Operation)
+	for _, log := range logs {
+		sociLog := SociLog{}
+		if log != "" {
+			err := json.Unmarshal([]byte(log), &sociLog)
+			if err != nil {
+				return err
+			}
+
+		}
+		if sociLog.Msg == sociLogsFuseMessage {
+			var tempOperation BaseOperation
+			err := json.Unmarshal([]byte(log), &tempOperation)
+			if err != nil {
+				return err
+			}
+			op := tempOperation.Operation + tempOperation.Path
+
+			val, ok := m[op]
+
+			if ok {
+				val.Count = val.Count + 1
+				m[op] = val
+
+			} else {
+				m[op] = Operation{
+					Operation:                 tempOperation.Operation,
+					Path:                      tempOperation.Path,
+					FirstAccessTimeAfterStart: tempOperation.Time.Sub(fa.ContainerStartTime).String(),
+					Count:                     1,
+				}
+			}
+			count, ok := fa.TotalOperationCount[tempOperation.Operation]
```

Comment:
nit: maps in go return the value type's zero value if not present which is 0 for integers. It should be safe to skip these checks and just do:
```
fa.TotalOperationCount[tempOperation.Operation]++
```

If it's not there it will add 1 to 0 and store it which is what we want.